x/evm/types/errors.go

```diff
@@ -75,12 +79,33 @@ var (
 
 // NewExecErrorWithReason unpacks the revert return bytes and returns a wrapped error
 // with the return reason.
-func NewExecErrorWithReason(revertReason []byte) error {
-	hexValue := hexutil.Encode(revertReason)
-	reason, errUnpack := abi.UnpackRevert(revertReason)
+func NewExecErrorWithReason(revertReason []byte) *RevertError {
```

Comment:
we should create a few unit tests for this with some of the example inputs

Reply:
test added